packages/storage-ui/src/Components/Modules/AddCIDModal.tsx

```diff
@@ -63,97 +75,136 @@ interface IAddCIDModuleProps {
 
 const AddCIDModal = ({ modalOpen = false, close }: IAddCIDModuleProps) => {
   const classes = useStyles()
-  const { addPin, refreshPins } = useStorage()
-  const inputRef = useRef<any>(null)
+  const { addPin, refreshPins, searchCid } = useStorage()
   const [accessingCID, setAccessingCID] = useState(false)
+  const [cidError, setCidError] = useState("")
+  const [cid, setCid] = useState("")
+  const [showWarning, setShowWarning] = useState(false)
 
-  const onSubmit = useCallback((values, helpers) => {
-    helpers.setSubmitting(true)
+  const onClose = useCallback(() => {
+    setCid("")
+    setShowWarning(false)
+    setCidError("")
+    close()
+  }, [close])
+
+  const onSubmit = useCallback(() => {
     setAccessingCID(true)
-    addPin(values.cid.trim())
+    addPin(cid.trim())
       .then(() => {
-        helpers.resetForm()
-        close()
+        onClose()
       })
       .catch((e) => {
-        helpers.setFieldError("cid", e.message)
         console.error(e)
       })
       .finally(() => {
         refreshPins()
         setAccessingCID(false)
-        helpers.setSubmitting(false)
       })
-  }, [addPin, close, refreshPins])
+  }, [addPin, cid, onClose, refreshPins])
+
+  const warnforDuplicatesCid = useCallback(() => {
+    searchCid(cid)
+      .then((res) => {
+        if(res && res.results?.length){
+          setShowWarning(true)
+        }
+      })
+  }, [cid, searchCid])
+
+  const onCidChange = useCallback((cid?: string | number) => {
+    setCidError("")
+    setShowWarning(false)
+
+    cid && setCid(cid?.toString())
```

Comment:
This is preventing erasing the first character of the input. This should fix it.
```suggestion
    setCid(cid?.toString() || "")
```